wiki/views/article.py

```diff
@@ -401,6 +401,80 @@ def get_context_data(self, **kwargs):
         return super(Edit, self).get_context_data(**kwargs)
 
 
+class Move(FormView, ArticleMixin):
+
+    form_class = forms.MoveForm
+    template_name= "wiki/move.html"
+
+    @method_decorator(get_article(can_write=True, not_locked=True))
+    def dispatch(self, request, article, *args, **kwargs):
+        return super(Move, self).dispatch(request, article, *args, **kwargs)
+
+    def get_initial(self):
+        initial = FormView.get_initial(self)
+        return initial
+
+    def get_form(self, form_class):
+        """
+        Checks from querystring data that the edit form is actually being saved,
+        otherwise removes the 'data' and 'files' kwargs from form initialisation.
+        """
+        kwargs = self.get_form_kwargs()
+        return form_class(**kwargs)
+
+    def get_context_data(self, **kwargs):
+
+        kwargs['root_path'] = models.URLPath.root()
+
+        return super(Move, self).get_context_data(**kwargs)
+
+    def form_valid(self, form):
```

Comment:
Maybe this should have an atomic decorator? While the re-parenting in mptt seems fairly atomic, there's a couple of other calls surrounding it.